timemachine/cpp/src/hilbert_curve.cu

```diff
@@ -0,0 +1,85 @@
+#include "gpu_utils.cuh"
+#include "hilbert_curve.hpp"
+#include "kernels/k_hilbert.cuh"
+#include "vendored/hilbert.h"
+#include <cub/cub.cuh>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace timemachine {
+
+HilbertCurve::HilbertCurve(int N)
```

Comment:
Would it make sense for `N` to be a template parameter to enable potential optimizations?

Reply:
Disregard, sorry. I had originally misinterpreted `N` as the size of the grid.

Reply:
N is dependent on the system size, so we couldn't make it a template parameter without some weirdness. By constant do you simply mean `const int N` as the input parameter? That is doable, though don't think it would make much difference since `N_` is `const`.

Reply:
Made the param `const` to be consistent with the `.hpp` definition of the constructor.